packages/kilt-did/README.md

```diff
@@ -3,6 +3,339 @@
+# Kilt DIDs
 
-# Kilt Did support
+A KILT Decentralised Identifier (DID) is a string uniquely identifying each KILT user. A DID can be thought of as a container of different keys that are all under the control of the same DID subject (see the [DID Core spec](https://www.w3.org/TR/did-core/) for more information).
+
+The simplest type of KILT DID is a **light DID**, called this way because it can be generated and used offline without requiring any Internet connection (hence any connection with the KILT blockchain at all). Although very cheap, light DIDs are not very flexible and are suitable for lower-security use cases. In more complex use cases, a **full DID** is more indicated, which allows the subject to store several different keys (and key types) and replace them over time, with the help of the KILT blockchain.
+
+## Light DIDs
+
+An example of a light KILT DID is the following:
+
+```
+did:kilt:light:014sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar
+```
+
+Beyond the standard prefix `did:kilt:`, the `light:` component indicates that this DID is a light DID, hence it can be resolved and utilized offline.
+
+Light DIDs optionally support the specification of an **encryption key** (of one of the supported key types) and **service endpoints**, which are serialised and encoded and added at the end of the DID, like the following:
+
+```
+did:kilt:light:014sxSYXakw1ZXBymzT9t3Yw91mUaqKST5bFUEjGEpvkTuckar:omFlomlwdWJsaWNLZXlYID9hc7PRyRlUp+syykH3KrsVZlObWlfqtegO1KRzuo8zZHR5cGV4GHgyNTUxOS14c2Fsc2EyMC1wb2x5MTMwNWFzgaNiaWRlZW1haWxvc2VydmljZUVuZHBvaW50bG15QGVtYWlsLm9yZ2R0eXBlbEVtYWlsU2VydmljZQ==
+```
+
+### Creating a light DID
+
+To create a light DID, there needs to be a keystore instance that conforms to the [Keystore interface](../types/src/Keystore.ts). For the sake of ease of use, this package includes a [demo keystore](./src/DemoKeystore/DemoKeystore.ts) which can be used to generate key pairs that are kept in memory and disappear at the end of the program execution.
+
+> **Using the demo keystore in production is highly discouraged as all the keys are kept in the memory and easily retrievable by malicious actors.**
+
+The following is an example of how to create a light DID after creating an instance of the demo keystore.
+
+```typescript
+import {
+  DemoKeystore,
+  EncryptionAlgorithms,
+  LightDidDetails,
+  SigningAlgorithms,
+} from '@kiltprotocol/did'
+import type { IServiceDetails } from '@kiltprotocol/types'
+
+// Instantiate the demo keystore.
+const keystore = new DemoKeystore()
+
+// Generate seed for the authentication key/
+const authenticationSeed = '0x123456789'
+
+// Ask the keystore to generate a new keypair to use for authentication with the generated seed.
+const authenticationKeyPublicDetails = await keystore.generateKeypair({
+  alg: SigningAlgorithms.Ed25519,
+  seed: authenticationSeed,
+})
+
+// Create a light DID from the generated authentication key.
+const lightDID = new LightDidDetails({
+  authenticationKey: {
```

Comment:
C: This could maybe in the future just take the `authenticationKeyPublicDetails` and extract the information itself

Reply:
I agree. Or even just a KILT account, if we re-introduce that for blockchain addresses.